### DISCUSSION TITLE
What to do with the Strava implementation

### DISCUSSION THREAD

**Discussion by JaapvanEkris** [JaapvanEkris/openrowingmonitor#64] [Polls]
Now that our file creation is reasonably complete, I'm looking at what to do with these files. Lars has created a Strava integration 3 years ago as part of 0.8.2, which I maintained since then. But that remained the only integration, as Strava used to be interesting as a single portal to many other analysis platforms, greatly increasing data portability for users in a single integration.

However, today [Strava announced these platforms aren't allowed to analyse data anymore](https://analytics.rowsandall.com/2024/12/03/strava-api-changes-affecting-rowsandall/), greatly limiting your use of your own data. RowsAndAll.com is one of the affected parties, which can't do analysis on rowing data obtained via Strava, nor can coaches (which you pay to help you) see your data anymore. This renders Strava useless as a conduit. In all honesty, I have strong issues with their approach, as it goes against OpenRowingMonitor's founding principle that your data must be as free and accessible for you as you want it to be. I think that should be the leading principle for the entire sport metrics ecosystem.

Some things also bother me from a technical perspective, as I see OpenRowingMonitor  somehow decides that the user gets a question, and that triggers the Strava upload in a complex interaction between frontend and backend. That implementation doesn't fit well with our new technical architecture, and it will never work on headless units. Given that Strava uses Oauth2 authentication, I have concerns it will ever fit well in a way that Oauth2 is intended while fitting our intended use cases and architecture.

I also see what other (free!) data analysis platforms, Intervals.icu and RowsAndAll being the primary ones, are doing. Automatic uploads are much simpler, their analysis is far superior to Strava's (especially for rowing) and they do respect the user's desire to share and analyse data freely across many platforms. I personally think eliminating Strava's role as a conduit here would benefit users, as it puts users firmly in control of who sees their data, and not a vague social media company that struck gold.

Personally, as a daily rower on EXR, I do post to Strava for social reasons, but I use EXR to connect to Strava as it also uploads nice pictures and virtual geodata. This, combined with Strava's diminished role as (conduit to) a data analysis platform, would make direct Strava integration for OpenRowingMonitor less relevant.

So my key question is if we want to keep Strava integration and how elaborate should it be?

**Comment by Frits1980** [comment]
I think a lot of people will use the build in health / fitness apps in Android or Apple ecosystem. From a political standpoint I would prefer not to, but that is irrelevant to what is a wise choice for ORM to integrate with. And since these apps from Apple or Android can sync data with a lot of other apps that brings the most versatile options for users. But I don’t know if these apps from Apple nd Android support rowing.

**Reply by JaapvanEkris** [reply]
> I think a lot of people will use the build in health / fitness apps in Android or Apple ecosystem.

Do they? I don't know anyone who does. Their interoperability is extremely bad and their data analysis is even worse. Most rowers I know are firmly into the Garmin or Polar ecosystems. And people serious about training are typically more geared towards intervals.icu or TrainingPeaks.

And both platforms are an absolute nightmare to integrate with, as they blatenly disregard any industry standard out there. Most apps recognise the PM5 as a BLE device, so that is the only way of getting data in.

> From a political standpoint I would prefer not to, but that is irrelevant to what is a wise choice for ORM to integrate with. And since these apps from Apple or Android can sync data with a lot of other apps that brings the most versatile options for users. But I don’t know if these apps from Apple nd Android support rowing.

In all honesty, these platforms suck for rowing. As does Strava, especially for analysis. Strava used to be interesting as a springboard for other platforms, for further analysis. However, the above mentioned approach of blocking independent data analysis platforms, and their recent acquisition of analysis/planning platform Runna, suggests they won't get much more open.

Lately we see a huge shift towards intervals.icu as it is open, free and allows automatic upload to various platforms. So in 0.9.6 we follow and allow automatic upload to them. Another is RowsAndAll.com, which is becoming the industry standard for rowing data anlysis, which also is free (with premium subscription for more data), and also is a good conduit to other platforms. In 0.9.6 that is also an integrated feature.

**Comment by JaapvanEkris** [comment]
For all interested: the current new-ble-api branch, which will become 0.9.6 in a couple of weeks is functional and will contain the redesigned Strava API (as well as support for intervals.icu and RowsAndAll.com). This will be a breaking change: you will need to authorize the Strava app manually (procedure will be described in detail in the manual, but it will be based on https://gist.github.com/michaellihs/bb262e2c6ee93093485361de282c242d).

We will support two types of upload (as for all three services):
* Automatic: as soon as the workout ends, it will be uploaded automatically,
* Manual: you will need to push the button on the screen to upload

**Comment by JaapvanEkris** [comment]
[Strava has updated their new API requirements](https://communityhub.strava.com/insider-journal-9/an-update-to-our-developer-program-13428), essentially requiring people to subscribe to use the API. As I personally don't see any value in Strava, I will not be buying one. As such I will not be able to test or fix the Strava implementation. I will leave it in as-is, but if it breaks, I can't repair it.

In June 2027 Strava will introduce several breaking changes which require significant development and test work. As things are currently (as: no alternative developer picking this up), this will mark the end of the line for integrated Strava interface for OpenRowingMonitor. If possible we will provide a setup for a shell script, depending in available libraries. But it will not be part of OpenRowingMonitor.
